### PR TITLE
Emulator build without Relic + Enable crypto no-Relic tests

### PR DESCRIPTION
**Pull request by tarakby**
- Crypto package: remove legacy threshold signature functions from consensus voting V1 
- Crypto package and `module/signature` package: add public functions to mirror all BLS-related functions, the new function are compiled without the relic tag and panic at runtime. These functions are only added to allow build of projects that do not require BLS (emulator for instance)
- Crypto: add tests to make sure new functions panic as expected.
- CI: add a new test that checks if the FVM is build-able without the "relic" tag. This test is required to make sure the emulator can build. 
- CI :crypto package Makefile: enable crypto tests without the relic build tag in CI (removed by mistake ?!)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -79,8 +79,13 @@ unittest: unittest-main
 	$(MAKE) -C crypto test
 	$(MAKE) -C integration test
 
+.PHONY: emulator-build
```
Comment (tarakby) [inline Makefile:82]: New CI test that runs no Go tests, but just checks the FVM can be built

File: crypto/Makefile [hunk 2]
```diff
@@ -7,11 +7,13 @@ ADX_SUPPORT := $(shell if ([ -f "/proc/cpuinfo" ] && grep -q -e '^flags.*\badx\b
 
 .PHONY: test
 test:
-	# test all packages with Relic library enabled
+	# test all packages with Relic library disabled
```
Comment (tarakby) [inline crypto/Makefile:10]: This test should be run by CI, but it looks like it was removed!

File: crypto/Makefile [hunk 3]
```diff
@@ -7,11 +7,13 @@ ADX_SUPPORT := $(shell if ([ -f "/proc/cpuinfo" ] && grep -q -e '^flags.*\badx\b
...
+	GO111MODULE=on go test -coverprofile=$(COVER_PROFILE) $(if $(TEST_LONG),,-short) $(if $(JSON_OUTPUT),-json,) $(if $(NUM_RUNS),-count $(NUM_RUNS),) $(if $(VERBOSE),-v,) ./...
+	# test BLS-related packages (with Relic library enabled)
 ifeq ($(ADX_SUPPORT), 1)
-	GO111MODULE=on go test -coverprofile=$(COVER_PROFILE) $(if $(TEST_LONG),,-short) $(if $(JSON_OUTPUT),-json,) $(if $(NUM_RUNS),-count $(NUM_RUNS),) --tags relic $(if $(VERBOSE),-v,) ./...
+	GO111MODULE=on go test -coverprofile=$(COVER_PROFILE) $(if $(TEST_LONG),,-short) $(if $(JSON_OUTPUT),-json,) $(if $(NUM_RUNS),-count $(NUM_RUNS),) --tags relic $(if $(VERBOSE),-v,)
```
Comment (tarakby) [inline crypto/Makefile:14]: no need to run relic tests in sub directories (BLS is only implemented in the root directory)

File: crypto/thresholdsign.go [hunk 4]
```diff
@@ -143,138 +139,3 @@ func IsNotEnoughSharesError(err error) bool {
 	var target *notEnoughSharesError
 	return errors.As(err, &target)
 }
-
-// ReconstructThresholdSignature is a stateless api that takes a list of
```
Comment (tarakby) [inline crypto/thresholdsign.go:147]: V1 legacy functions

**Review by janezpodhostnik — APPROVED** [review]
LGTM. 

I like the CI addition a lot.

File: crypto/bls_no_relic.go [hunk 5]
```diff
@@ -0,0 +1,124 @@
+//go:build !relic
+// +build !relic
```
Comment (sideninja) [inline crypto/bls_no_relic.go:2]: Couldn't we remove this since the minimum Go version is 1.16 where this is no longer supported?

**Review by sideninja — APPROVED** [review]
LGTM 🚀  Great CI check

File: crypto/bls_no_relic.go (hunk 5 above)
Comment (tarakby) [reply crypto/bls_no_relic.go:2]: At least one of the 2 lines is required to mention that this file is only compiled when the build doesn't contain the "relic" tag. 
The extra line is added by my IDE, although I'm using Go 1.17.5.

**Comment by tarakby** [comment]
bors merge

**Comment by tarakby** [comment]
bors merge

**Pull request merged**